### PR TITLE
Modify base codec to use io.{Reader,Writer} instead of slices

### PR DESCRIPTION
**Pull request by Kubuxu**

### CODE REVIEW

**Comment by Kubuxu** [comment]
Can you also rebase it on top of master so codecov runs.

File: base/b64/codec.go [hunk 1]
```diff
@@ -21,13 +21,12 @@ type decoder struct {
 }
 
 func (d decoder) Decode(v interface{}) error {
```
Comment (whyrusleeping) [inline base/b64/codec.go]: this feels weird. Why accept an interface if we're forcing a type cast on it?

Comment (Kubuxu) [inline base/b64/codec.go]: Because that is the Multicodec interface.

Comment (Kubuxu) [inline base/b64/codec.go]: https://github.com/multiformats/go-multicodec/blob/master/coding.go#L11-L30

Comment (whyrusleeping) [inline base/b64/codec.go]: Oh.... treating encodings as codecs feels very wrong to me.  The interface doesnt mesh well, and its a separate step altogether.

Theres a reason that cids have the base encoding as a separate field from the codec

Comment (Kubuxu) [inline base/b64/codec.go]: True, but otherwise I can't use existing abstractions like mux codec and so on.

**Comment by mvdan** [comment]
With permission from the rest of the IPLD team, I'm reviving this Go module to simply expose the multicodec table as code-generated Go constants. I'll be deleting all existing code in a commit (to preserve history) and will tag v0.2.0 once the new, simpler package is ready.

As such, I'm closing all previous issues. Thanks!

**Pull request closed without merging**